skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,350 @@
+        self.assertTrue(f1 < f2)
+        self.assertTrue(f1 <= f2)
+        self.assertFalse(f1 > f2)
+        self.assertFalse(f1 >= f2)
+
+    def test_contains(self):
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertTrue('name' in f1)
+        self.assertTrue('function' in f1)
+        self.assertFalse('gene' in f1)
+
+    def test_hash(self):
+
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        f2 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertEqual(hash(f1), hash(f2))
+
+    def test_drop(self):
+        im = IntervalMetadata()
+        im.add(metadata={'gene': 'sagA', 'location': 0},
+               intervals=[(0, 2), (4, 7)])
+        self.assertEqual(im._metadata[0].intervals,
+                         [(0, 2), (4, 7)])
+        self.assertEqual(im._metadata[0].metadata,
+                         {'gene': 'sagA', 'location': 0})
+        self.assertTrue(im._intervals is not None)
+
+        feats = im.query(intervals=[(1, 2)])
+
+        self.assertEqual(len(feats), 1)
+        self.assertEqual(feats[0].metadata, {'gene': 'sagA', 'location': 0})
+
+        feats[0].drop()
+        feats = im.query(intervals=[(1, 2)])
+        self.assertEqual(len(feats), 0)
+
+    def test_equal(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2), (4, 7)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+
+        f2 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2), (4, 8)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+
+        f3 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2), (4, 8)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagB', 'function': 'transport'})
+        self.assertEqual(f, f1)
+        self.assertNotEqual(f, f2)
+        self.assertNotEqual(f, f3)
```

Comment:
also provide the cases that diff in `metadata` or `boundaries` leads to unequality?